src/main/java/io/getunleash/repository/FeatureRepositoryImpl.java

```diff
@@ -201,4 +245,29 @@ public WasmResponse<VariantDef> getVariant(String toggleName, UnleashContext con
     public Stream<FeatureDefinition> listKnownToggles() {
         return this.engine.listKnownToggles().stream().map(FeatureDefinition::new);
     }
+
+    public void handleStreamingUpdate(String data) {
+        try {
+            engine.takeState(data);
+            // TODO: write backup when engine exposes current stage
```

Comment:
yggdrasil needs to expose current state we can store to a backup file. Same problem we had in the Ruby SDK

Reply:
I will pair with @sighphyre next week to think how to expose full internal flag state from ygg so that we can store it. Node SDK is doing exactly that and using existing backup mechanism (but not using Ygg yet).